api/v1alpha1/topolvmcluster_types.go

```diff
@@ -28,8 +30,62 @@ type TopoLVMClusterSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of TopoLVMCluster. Edit topolvmcluster_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// DeviceSelector is a set of rules that should match for a device to be included in this TopoLVMCluster
+	// +Optional
+	DeviceClasses []DeviceClass `json:"deviceClasses,omitempty"`
+}
+
+type DeviceClass struct {
+	// Name of the class, the VG and possibly the storageclass.
+	Name string `json:"name,omitempty"`
+	// DeviceSelector is a set of rules that should match for a device to be included in this TopoLVMCluster
+	// +optional
+	DeviceSelector *DeviceSelector `json:"deviceSelector,omitempty"`
+	// NodeSelector chooses nodes
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// Config for this deviceClass, lvm settings are a field here
+	// +optional
+	Config *DeviceClassConfig `json:"config,omitempty"`
+}
+
+// DeviceSelector matches if all the rules match.
+// (ruleType1 AND ruleType2 AND ... AND ruleTypeN-1 AND ruleTypeN)
+// Each characteristic is represented either by a ruleType, a ruleType is either a list of OR'd rules or a single rule
+type DeviceSelector struct {
+	// MaxMatchesPerNode to allow a use-cases where not all matches on a node are consumed.
+	// +optional
+	MaxMatchesPerNode *int `json:"maxMatchesPerNode,omitempty"`
+	// Devices is the list of devices that should be used for automatic detection.
+	// This would be one of the types supported by the local-storage operator. Currently,
+	// the supported types are: disk, part. If the list is empty only `disk` types will be selected
+	// +optional
+	DeviceTypes []DeviceType `json:"deviceTypes,omitempty"`
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperties []DeviceMechanicalProperty `json:"deviceMechanicalProperties,omitempty"`
```

Comment:
`DeviceMechanicalProperties`
If the attribute is just a boolean, why to use a slice of strings...? (If not provided we can select both types of disk.)

Reply:
We could select only-hdd or only-sdd or both with this setting